common/src/main/java/com/wynntils/features/user/GuildStarsToRankFeature.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import com.wynntils.wynn.objects.GuildRank;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class GuildStarsToRankFeature extends UserFeature {
+
+    private static final Pattern guildMessage = Pattern.compile("§3\\[§b(★*)§3(.+)§3\\] (.*)");
```

Comment:
You don't need to to this. The ChatManager already separates out and tags guild messages. You are just listening to a "raw" event, instead of the proper "cooked" one.

Reply:
Does that capture stars name abd message? This just makes it easier

Reply:
What does `abd message` mean?

Reply:
> What does `abd message` mean?

meant and

Reply:
since this regex captures stars in one group username in one group and message in one group. just for convenience tbh

Reply:
there's no reason you can't apply a (simplified from this) guild rank regex to messages that have already been properly filtered as guild messages

Reply:
either way i do not know how to simplify this since i capture everything impotant to me